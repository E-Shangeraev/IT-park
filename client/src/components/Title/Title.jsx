import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import './Title.scss'

const Title = ({ children, aura }) => (
  <h2 className={classNames('title', { 'title--aura': aura })}>{children}</h2>
)

Title.propTypes = {
  children: PropTypes.node.isRequired,
  aura: PropTypes.bool,
}

Title.defaultProps = {
  aura: false,
}

export default Title
