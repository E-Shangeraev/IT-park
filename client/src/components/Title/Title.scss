@import '../../scss/variables.scss';

.title {
  position: relative;
  margin: 0;
  font-size: 44px;
  line-height: 150%;
  z-index: 1;

  b {
    color: $orange;
  }
  span {
    display: block;
    position: relative;
  }
  span:nth-of-type(2) {
    left: 6%;
  }
  &::before {
    position: absolute;
    content: '';
    top: 0;
    left: 0;
    width: 6.8em;
    height: 6.8em;
    opacity: 0.5;
    transform: translate(-50%, -40%);
    border-radius: 50%;
    background: radial-gradient($orange 1%, transparent 50%) no-repeat;
    z-index: -1;
  }
}

@media (max-width: 1440px) {
  .title {
    font-size: 40px;
  }
}
@media (max-width: 1200px) {
  .title {
    font-size: 36px;
  }
}
@media (max-width: 1024px) {
  .title {
    font-size: 30px;
  }
}
@media (max-width: 768px) {
  .title {
    font-size: 24px;
  }
}
@media (max-width: 500px) {
  .title {
    font-size: 18px;
  }
}
@media (max-width: 320px) {
  .title {
    font-size: 16px;
  }
}
