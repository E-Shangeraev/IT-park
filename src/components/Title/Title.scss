@import '../../scss/variables.scss';

.title {
  position: relative;
  margin: 0;
  font-size: 44px;
  line-height: 150%;
  z-index: 1;

  b {
    color: $orange;
  }
  span {
    display: block;
    position: relative;
  }
  span:nth-of-type(2) {
    left: 6%;
  }
  &::before {
    position: absolute;
    content: '';
    top: 0;
    left: 0;
    width: 300px;
    height: 300px;
    opacity: 0.5;
    transform: translate(-50%, -40%);
    border-radius: 50%;
    background: radial-gradient($orange 1%, transparent 50%) no-repeat;
    z-index: -1;
  }
}
