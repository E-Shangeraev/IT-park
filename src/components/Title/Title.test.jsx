/* eslint-disable no-undef */
import React from 'react'
import Title from './Title'

describe('Title component', () => {
  it('should render Title component with props', () => {
    const component = shallow(<Title>Test</Title>)
    expect(component).toMatchSnapshot()
  })
})
