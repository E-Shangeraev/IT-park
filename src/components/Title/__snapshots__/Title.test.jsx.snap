// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should render Title component with props 1`] = `
<h2
  className="title"
>
  Test
</h2>
`;
